src/irs/core/interrogator.py

```diff
@@ -328,5 +178,46 @@ async def interrogate(self, *, payload: event_schemas.FileUploadReceived):
                 s3_endpoint_alias=payload.s3_endpoint_alias,
             )
 
-        # Everything has been processed successfully, add fingerprint to db for lookup
-        await self._fingerprint_dao.insert(fingerprint)
+    async def remove_staging_object(
+        self, *, payload: event_schemas.FileInternallyRegistered
```

Comment:
The event schemas are an adaptor detail that is enforced by the producer of the events. The core of a consuming microservice should, however, feel free to define its interface independent of the surrounding infrastructure. This is a core principle of hexagonal architecture. Thus I would propose not to use event schemas (or anything that indicates the event-driven nature of the services interface) in the core. This might be better:

```suggestion
        self, *, file_id: str, endpoint_alias: str
```

Reply:
The main interrogate functionality also uses the payload directly, wanted to address this in the refactoring PR.

Reply:
Sounds good. Let's do it in a separate PR.